diff -wpruN '--exclude=*.orig' a~/src/nxt_router.c a/src/nxt_router.c
--- a~/src/nxt_router.c	1970-01-01 00:00:00
+++ a/src/nxt_router.c	1970-01-01 00:00:00
@@ -3692,7 +3692,7 @@ nxt_router_thread_exit_handler(nxt_task_
     nxt_event_engine_t   *engine;
     nxt_thread_handle_t  handle;
 
-    handle = (nxt_thread_handle_t) obj;
+    handle = (nxt_thread_handle_t)(uintptr_t) obj;
     link = data;
 
     nxt_thread_wait(handle);
diff -wpruN '--exclude=*.orig' a~/src/nxt_unit.c a/src/nxt_unit.c
--- a~/src/nxt_unit.c	1970-01-01 00:00:00
+++ a/src/nxt_unit.c	1970-01-01 00:00:00
@@ -3707,7 +3707,7 @@ nxt_unit_shm_open(nxt_unit_ctx_t *ctx, s
     char             name[64];
 
     snprintf(name, sizeof(name), NXT_SHM_PREFIX "unit.%d.%p",
-             lib->pid, (void *) pthread_self());
+             lib->pid, (void *) (uintptr_t)pthread_self());
 #endif
 
 #if (NXT_HAVE_MEMFD_CREATE)
